### DISCUSSION TITLE
Disabling extra heap_pages allocation for parachain clients

### DISCUSSION THREAD

**Discussion by crystalin** [paritytech/substrate#14232] [General]
Parachain blocks are validated on the relay using a wasmtime executor with fixed heap pages.
However it is possible to change the heap pages using onchain storage https://github.com/paritytech/substrate/blob/f4cb21c876d4ce19eb9ef09c39bebf83ba01d16d/primitives/state-machine/src/backend.rs#L447-L452

I would suggest to remove the heap pages support for parachain, preventing the case where a parachain block is produced with a given heap pages count that cannot be executed on the relaychain.

@bkchr What do you think ?

**Comment by bkchr** [comment]
I had though that I already had written an issue for it, but I was wrong :P 

Here it is now: https://github.com/paritytech/cumulus/issues/2641

**Reply by crystalin** [reply]
Sorry forgot to search also in cumulus :(

**Reply by bkchr** [reply]
Haha, the issue didn't existed before you asked :D
